horreum-web/src/Login.tsx

```diff
@@ -38,12 +38,18 @@ export default function LoginModal(props: LoginModalProps) {
                 onClick={() => {
                     setCreating(true)
                     store.dispatch({type: BASIC_AUTH, username, password});
-
-                    // TODO: instead of fetching userdata, should be fetching the user roles instead
-                    userApi.info([username || ''])
-                        .then(userdata => {
-                                alerting.dispatchInfo("LOGIN", "Log in successful", "Successful log in of user " + userdata[0].username, 3000)
-                                store.dispatch({type: UPDATE_ROLES, authenticated: true, roles: [/* TODO: the roles fetched */]});
+                    
+                    userApi.getRoles()
+                        .then(roles => {
+                                alerting.dispatchInfo("LOGIN", "Log in successful", "Successful log in of user " + username, 3000)
+                                store.dispatch({type: UPDATE_ROLES, authenticated: true, roles: roles});
+                                userApi.searchUsers(username!).then(
+                                    userData => store.dispatch({type: STORE_PROFILE, profile: userData.filter(u => u.username == username).at(0)}),
+                                    error => alerting.dispatchInfo("LOGIN", "Unable to get user profile", error, 30000))
```

Comment:
Could we add to the error msg to tell the user what to do in this situation? something like ". please contact your system administrator"